firebase-firestore/src/testUtil/java/com/google/firebase/firestore/testutil/TestUtil.java

```diff
@@ -526,6 +545,36 @@ public static ByteString streamToken(String contents) {
     return ByteString.copyFrom(contents, Charsets.UTF_8);
   }
 
+  /**
+   * An implementation of TargetMetadataProvider that provides controlled access to the
+   * `TargetMetadataProvider` callbacks. Any target accessed via these callbacks must be registered
+   * beforehand via `setSyncedKeys()`.
+   */
+  public static class TestTargetMetadataProvider
```

Comment:
Consider making this just a top-level class (i.e. in its own source file).

Reply:
Done